pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -429,25 +429,33 @@ contract MultiRewards is IMultiRewards, IDistributor, ReentrancyGuard, MultiRewa
 
         getVault().manageUserBalance(ops);
 
+        _updateRewardRate(pool, rewarder, rewardsToken, reward);
+        rewardData[pool][rewarder][rewardsToken].lastUpdateTime = block.timestamp;
+        rewardData[pool][rewarder][rewardsToken].periodFinish = block.timestamp.add(
+            rewardData[pool][rewarder][rewardsToken].rewardsDuration
+        );
+        emit RewardAdded(address(rewardsToken), reward);
+    }
+
+    function _updateRewardRate(
+        IERC20 pool,
+        address rewarder,
+        IERC20 rewardsToken,
+        uint256 newReward
```

Comment:
I find the addition of this function slightly odd, as we're altering the same data structure both inside it and on its callsite.

I can it it being potentially motivated by how verbose all of this is due to repeated access to `rewardData[pool][rewarder][rewardsToken]`, and I propose we do the following instead: keep the original code as-is (just changing the fixed point mul for regular mul), and then in a separate PR use storage pointers to the struct to a) reduce verbosity and b) save gas by computing the storage slot only once.

Reply:
Went ahead and implemented this myself in the interest of time.